src/Integration.UnitTests/Exclusions/ExclusionSettingsFileStorageTests.cs

```diff
@@ -0,0 +1,188 @@
+
+            settings.Should().BeNull();
+            logger.AssertOutputStrings(1);
+            logger.AssertOutputStringExists(string.Format(NotFoundMessage, projectKey));
+        }
+
+        [TestMethod]
+        public void GetSettings_HaveError_ReturnsNull()
+        {
+            var projectKey = "projectKey";
+            var filePath = GetFilePath(projectKey);
+
+            var file = new Mock<IFile>();
+            file.Setup(f => f.Exists(filePath)).Returns(true);
+
+            var logger = new TestLogger();
+
+            var fileSystem = CreateFileSystem(file.Object);
+
+            var testSubject = CreateTestSubject(fileSystem, logger);
+
+            var settings = testSubject.GetSettings(projectKey);
+
+            settings.Should().BeNull();
+            logger.AssertOutputStrings(1);
+            logger.AssertOutputStringDoesNotExist(string.Format(NotFoundMessage, projectKey));
+        }
+
+        [TestMethod]
+        public void SaveSettings_NoError_SavesSettings()
+        {
+            var projectKey = "projectKey";
+            var filePath = GetFilePath(projectKey);
+
+            var file = new Mock<IFile>();
+
+            var logger = new TestLogger();
+
+            var fileSystem = CreateFileSystem(file.Object);
+
+            var testSubject = CreateTestSubject(fileSystem, logger);
+
+            var settings = new ServerExclusions
+            {
+                Inclusions = new string[] { "inclusion1", "inclusion2" },
+                Exclusions = new string[] { "exclusion" },
+                GlobalInclusions = new string[] { "globalInclusion" },
+                GlobalExclusions = new string[] { "globalExclusion" }
+            };
+
+            testSubject.SaveSettings(projectKey, settings);
+
+            file.Verify(f => f.WriteAllText(filePath, objectJson));
+            logger.AssertOutputStrings(0);            
+        }
+
+        [TestMethod]
+        public void SaveSettings_Error_LogsError()
+        {
+            var projectKey = "projectKey";
+            var filePath = GetFilePath(projectKey);
+
+            var file = new Mock<IFile>();
+            file.Setup(f => f.WriteAllText(filePath, It.IsAny<string>())).Throws(new Exception());
+
+            var logger = new TestLogger();
+
+            var fileSystem = CreateFileSystem(file.Object);
+
+            var testSubject = CreateTestSubject(fileSystem, logger);
+
+            var settings = new ServerExclusions();          
+
+            testSubject.SaveSettings(projectKey, settings);
+
+            logger.AssertOutputStrings(1);
+            logger.AssertPartialOutputStringExists("Error writing settings for project");
+        }
+
+        private ExclusionSettingsFileStorage CreateTestSubject(IFileSystem fileSystem, ILogger logger)
+        {
+            return new ExclusionSettingsFileStorage(logger, fileSystem);
+        }
+
+        private IFileSystem CreateFileSystem(IFile file)
```

Comment:
Minor clean-up suggestion: the file system is never really checked, so maybe CreateTestSubject can receive IFile. Would make the tests slightly shorter and easier to read.